### ISSUE TITLE
segfault when reading compressed MPS file

### ISSUE THREAD

**Issue by mlubin** [JuliaOpt/GLPK.jl#14]
With https://github.com/mlubin/SimplexBenchmarks/blob/master/GenerateData/greenbea.gz:

```julia
julia> using GLPK

julia> lp = Prob()
Prob(Ptr{Void} @0x00000000045670d0)

julia> read_mps(lp, GLPK.MPS_FILE,"greenbea.gz")
Reading problem data from `greenbea.gz'...
Segmentation fault (core dumped)
```

Gdb trace:

```
#0  0x00007fffecde557c in crc32_little (len=<optimized out>, 
    buf=0x5810ac0 '*' <repeats 27 times>, "\n* SET UP THE INITIAL DATA *\n", '*' <repeats 27 times>, "\nNAME          GREENBEA", ' ' <repeats 58 times>, "\n*   Problem:\n*   ********\n*   An LP"..., 
    crc=<optimized out>) at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/crc32.c:259
#1  zlib_crc32 (crc=<optimized out>, buf=<optimized out>, len=len@entry=16384)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/crc32.c:218
#2  0x00007fffecdeba4a in zlib_inflate (strm=strm@entry=0x5e27ff8, flush=flush@entry=0)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/inflate.c:1232
#3  0x00007fffecde961a in gz_decomp (state=state@entry=0x5e27f80)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/gzread.c:189
#4  0x00007fffecde97f6 in gz_fetch (state=state@entry=0x5e27f80)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/gzread.c:245
#5  0x00007fffecde99d8 in zlib_gzread (file=0x5e27f80, buf=buf@entry=0x7fffffffcd8f, len=len@entry=1)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/gzread.c:339
#6  0x00007fffecde9ae3 in zlib_gzgetc (file=<optimized out>)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/zlib/gzread.c:398
#7  0x00007fffecd656db in z_fgetc (_fh=0x60096f0)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpenv07.c:580
#8  _glp_lib_xfgetc (fp=<optimized out>)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpenv07.c:214
#9  0x00007fffecda265f in read_char (csa=csa@entry=0x7fffffffd260)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpmps.c:167
#10 0x00007fffecda2f43 in indicator (csa=0x7fffffffd260, name=1)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpmps.c:214
#11 0x00007fffecda33ed in read_name (csa=0x7fffffffd260)
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpmps.c:395
#12 glp_read_mps (P=0x51c11a0, fmt=<optimized out>, parm=0x7fffffffcec0, fname=0x15cef20 "greenbea.gz")
    at /home/mlubin/.julia/GLPK/deps/src/glpk-4.52/src/glpmps.c:925
```

This used to work at some point a few months ago. If I uncompress the file, it seems to work fine.

**Comment by blegat** [comment]
This is fixed in 4.57 it seems
```julia
julia> using GLPK

julia> lp = Prob()
GLPK.Prob(Ptr{Void} @0x0000000003d909e0)

julia> read_mps(lp, GLPK.MPS_FILE, "greenbea.gz")
Reading problem data from 'data/greenbea.gz'...
Problem: GREENBEA
Objective: FAT0..J.
2393 rows, 5405 columns, 31499 non-zeros
19226 records were read
One free row was removed
0

julia> GLPK.version()
(4,57)
```
I guess we need to wait for the Windows binary to be updated because it is currently using 4.52.

**Comment by odow** [comment]
Closing as stale. Latest BinaryBuilder uses 4.64.
